### DISCUSSION TITLE
Sphinx Plot Directive: PyVista and Matplotlib

### DISCUSSION THREAD

**Discussion by adtzlr** [pyvista/pyvista#5951] [Q&A] [answered]
Hi! 

Do you know if it is possible to use the Sphinx plot directives of pyvista and matplotlib side-by-side? If I add both to my `conf.py`, I get an error. Any recommendations?

Thanks!

**Comment by tkoyama010** [comment, marked as answer]
@adtzlr PyVista supports Matplotlib plots in [CharMPL](https://docs.pyvista.org/version/stable/api/plotting/charts/_autosummary/pyvista.chartmpl#pyvista.ChartMPL). I recommend to plot Matplotlib results with the PyVista plot directives.
```python
import pyvista as pv
import numpy as np
import matplotlib.pyplot as plt

w = 3
Y, X = np.mgrid[-w:w:100j, -w:w:100j]
U = -1 - X**2 + Y
V = 1 + X - Y**2
speed = np.sqrt(U**2 + V**2)

f, ax = plt.subplots()
strm = ax.streamplot(X, Y, U, V, color=U, linewidth=2, cmap='autumn')
_ = f.colorbar(strm.lines)
_ = ax.set_title('Streamplot with varying Color')
plt.tight_layout()

chart = pv.ChartMPL(f)
chart.show()
```
![image](https://github.com/pyvista/pyvista/assets/7513610/558f0e37-7f7c-474f-abba-e18dfeb9acf0)

**Reply by adtzlr** [reply]
Thanks! I'll give it a try.

**Reply by adtzlr** [reply]
Is there any chance to import the plot-directives of matplotib and pyvista in `conf.py`? It seems to me that the Sphinx error comes from shared config variable names like `plot_include_source`.

I'm not too happy with the provided solution because I'd also have to add additional code blocks for matplotlib figures like:

```python
..  pyvista-plot::
    :context:
    
    >>> ax = some_mpl_ax()

..  pyvista-plot::
    :include-source: False
    :context:
    :force_static:
    
    >>> import pyvista as pv
    >>>
    >>> fig = ax.get_figure()
    >>> chart = pv.ChartMPL(fig)
    >>> chart.show()
```

Anyway, it works; your support is amazing @tkoyama010. You came up with a working solution in some minutes, thank you!!

**Reply by MatthewFlamm** [reply]
Can you share the error?  The documentation here mixes both, for example https://docs.pyvista.org/version/stable/examples/02-plot/image_depth.html.

**Reply by adtzlr** [reply]
Hi @MatthewFlamm! Thanks, but this is an example used by Sphinx Gallery. For *examples*, it works. I get an error if I try to enable both plot-directive extensions from matplotlib and pyvista to be used in the docstrings.

The plot-directive extensions in `conf.py`:
```python
extensions = [
    # ...,
    "pyvista.ext.plot_directive",
    "pyvista.ext.viewer_directive",
    "matplotlib.sphinxext.plot_directive",
]
```

```
make html
```

results in an error:
```
Extension error:
Config value 'plot_include_source' already present
```

Both directives use the same variable name, I think this is the problem here. Meanwhile, I use the solution from https://github.com/pyvista/pyvista/discussions/5951#discussioncomment-9175774.

**Reply by adtzlr** [reply]
Finally, it works. Thank you ❤️ . See https://docs.pyvista.org/extras/plot_directive.html for more details.
